sections/2-language-rules.md

```diff
@@ -645,3 +645,54 @@ package object dsl {
 }
 
 ```
+
+
+### 2.19 MUST NOT use `Seq.head`
+
+You might be tempted to this:
+
+```scala
+val userList: List[User] = ???
+
+// ....
+val firstName = userList.head.firstName
+```
+
+Don't ever do this, as this will throw `NoSuchElementException` if the sequence is empty.
+
+Alternatives:
+
+1. using `Seq.headOption` possibly combined with `getOrElse` or pattern matching
+
+    Example:
+    
+    ```scala
+    val firstName = userList.headOption match {
+        case Some(user) => user.firstName
+        case _ => "Unknown"
+      }
+    ```
+
+2. using pattern matching with the extractor `+:` 
+
+    Example:
+    
+    ```scala
+    val firstName = userList match {
+     case head +: _ => head.firstName
+     case _ => "Unknown"
+    }
+    ```
+
+3. using pattern matching with the cons operator `::` if you're dealing with a `List`
```

Comment:
N.B. this is the only sane way of using an extractor with the purpose of operating on `head`, the usage of a `List` 👍